csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -261,21 +235,17 @@ public void deactivate()
     }
 
     /// <summary>
-    /// Wait until the object adapter has deactivated.
-    /// Calling deactivate initiates object adapter
-    /// deactivation, and waitForDeactivate only returns when deactivation has been completed.
+    /// Wait until <see cref="deactivate" /> is called on this object adapter and all connections accepted by this
+    /// object adapter are closed. A connection is closed only after all outstanding dispatches on this connection have
+    /// completed.
     /// </summary>
     public void waitForDeactivate()
     {
         IncomingConnectionFactory[]? incomingConnectionFactories = null;
         lock (_mutex)
         {
-            //
-            // Wait for deactivation of the adapter itself, and
-            // for the return of all direct method calls using this
-            // adapter.
-            //
-            while ((_state < StateDeactivated) || _directCount > 0)
+            // Wait for deactivation of the adapter itself.
+            while (_state < StateDeactivated)
             {
                 Monitor.Wait(_mutex);
             }
```

Comment:
Is the if below still required?

Reply:
Why do you think it's no longer required?

Reply:
I was thinking the `if` condition would be always true after the while loop. But I was wrong, we can still be in the deactivated state